pennylane/ops/op_math/exp.py

```diff
@@ -279,9 +279,22 @@ def _recursive_decomposition(self, base: Operator, coeff: complex):
             ops = base.ops if isinstance(base, Hamiltonian) else base.operands
             return self._trotter_decomposition(ops, coeffs)
 
-        for op_class in has_unitary_generator_types:
-            # Check if the exponentiated operator is a generator of another operator
-            if op_class.num_wires in {base.num_wires, AnyWires} and op_class is not qml.PauliRot:
+        # Store operator classes with generators
+        has_generator_types = []
+        has_generator_types_anywires = []
+        for op_name in qml.ops.qubit.__all__:  # pylint:disable=no-member
+            op_class = getattr(qml.ops.qubit, op_name)  # pylint:disable=no-member
+            if op_class.has_generator:
+                if op_class.num_wires == AnyWires:
+                    has_generator_types_anywires.append(op_class)
+                elif op_class.num_wires == base.num_wires:
+                    has_generator_types.append(op_class)
+        # Ensure op_class.num_wires == base.num_wires before op_class.num_wires == AnyWires
+        has_generator_types += has_generator_types_anywires
+
+        for op_class in has_generator_types:
+            # PauliRot and PCPhase have different positional args
+            if op_class not in {qml.PauliRot, qml.PCPhase}:
```

Comment:
Question: why do we exclude `PCPhase` here? It certainly has a single unitary generator.

Reply:
The original decomposition procedure did not include `PCPhase` in the list of operator classes to iterate over here, but the explicit exclusion of `PauliRot` was already present. `PauliRot` requires an extra argument to instantiate, so the next line fails if it is not excluded. As `PCPhase` also has an extra positional argument it should cause the same error on the next line. I have just tried this however, and the tests do not fail if `PCPhase` is allowed; this means that the following line is never reached with `op_class == PCPhase`. I could allow `PCPhase`, test for that case and give different args for it, if we think that case should be handled here (it wasn't handled previously, so I wasn't sure if it was supposed to be possible).

Reply:
I see. These edge cases we can always think about later. For this PR (and unitaryhack) this is sufficient.